Draft of specification of stc_ columns

msdemlei opened the pull request:
These are tables making available the new metadata from VODataService 1.2. Adding these is not trivial, as we effectively require in-database MOC support for RegTAP with this. We could put them into a separate data model, but, really, I think MOC in SQL is such a great thing that I'd hope it'll be widely available in the near future anyway.

msdemlei committed:
Draft of specification of stc_ columns

mbtaylor commented on RegTAP.tex:
```diff
@@ -2239,6 +2252,152 @@ \subsection{The alt\_identifier Table}
+clauses) or in what form it will be returned; this is deferred to a
+future DALI version specifying the moc xtype.  Implementations MUST,
+however, evaluate the ADQL CONTAINS and INTERSECTS predicates with
+coverage as one argument and ADQL CIRCLEs and POLYGONs as the other.
```
Should CONTAINS/INTERSECTS be required to work for POINT as well?
The other thing that queries will presumably want to do is check whether a coverage element exists; I guess `IS NOT NULL` is good for that and sufficiently obvious not to need comment.

mbtaylor commented on RegTAP.tex:
```diff
@@ -2239,6 +2252,152 @@ \subsection{The alt\_identifier Table}
+clients are advised that on services supporting MOC literals, it is
+probably much faster and more exact to use MOC-MOC comparisons to query
+\rtent{coverage}.
+
```
The other new sections include the text  _"The ivoid column should be an explicit foreign key into resource."_ - should that be here too?

mbtaylor reviewed (commented):
As noted there are clearly implications for ease of implementation, I can't comment on those, though other likely implementors should do.  Other than that, this looks reasonable.

A couple of other items:
* In view of the upcoming MOC 2.0, does the characterisation of space+time coverage with a (spatial) MOC and a union-of-time-intervals, as opposed to an STMOC, need additional comment?  Given that this is just reflecting the content of VODataService 1.2 it probably has to be that way, and it's arguably more appropriate for the RegTAP discovery use case, so probably not.  But perhaps refer to "spatial MOCs" rather than just MOCs in the text?

* Should the (TAP_SCHEMA) Units for the {time,spectral}_{start,end} columns be mandated?

* If accepted some other updates will be needed in due course elsewhere in the document e.g. Figure 2.

msdemlei committed:
Minor editorial fixes in response to reviewer comments.

Specifically, https://github.com/ivoa-std/RegTAP/pull/3#pullrequestreview-870400755

mbtaylor reviewed (dismissed):
OK by me as far as the text goes; but potential implementors will have to comment about whether this is acceptable.

tsdower commented:
Thanks, @mbtaylor. I have implementation concerns, but haven't had a chance to dig into that part, and stalled out on even checking the proposal text itself.

It seems like we either need in-db MOC support on our infrastructure or, potentially, external service results filtering. Both of these imply a great deal of work, and will need to be considered at a later date.

msdemlei commented:
On Wed, Feb 02, 2022 at 04:14:25AM -0800, Mark Taylor wrote:
> @mbtaylor commented on this pull request.
> more appropriate for the RegTAP discovery use case, so probably
> not.  But perhaps refer to "spatial MOCs" rather than just MOCs in
> the text?

It's "spatial MOC" now in the introduction.  Since it's been brought
up before, it might be prudent to be a bit more defensive, in
particular since as you say, the Registry might not be the ideal
proving ground for STE-MOCs; it will be hard enough to rope in more
publishers to provide plain old MOCs and separate time and energy
coverage.

But really, even if I wanted, given that SMOC support in postgres is
still rather bleeding-edge and we can't even put them into a VOTable,
it's a tooling issue: if we want to get this out of the door within
the next few years, it's probably unwise to go for STEMOCs.

> * Should the (TAP_SCHEMA) Units for the {time,spectral}_{start,end}
> columns be mandated?

Indeed.

> * If accepted some other updates will be needed in due course
> elsewhere in the document e.g. Figure 2.

True.  But since the layout on this is tricky, I'd like to postpone
the fiddling until I'm reasonably sure the stc tables won't go to a
separate document.

But since it's quick, I *have* added MOC into RegTAP's architecture
diagram.

> +future DALI version specifying the moc xtype.  Implementations MUST,
> +however, evaluate the ADQL CONTAINS and INTERSECTS predicates with
> +coverage as one argument and ADQL CIRCLEs and POLYGONs as the other.
> Should CONTAINS/INTERSECTS be required to work for POINT as well?

Ouch, yes, CONTAINS(POINT, coverage) needs to work, of course.
Anything else with points I'd not require.  The INTERSECT decay to
CONTAINS I think we'll get rid of in ADQL 2.1.  Or at least I hope
so.

> The other thing that queries will presumably want to do is check
> whether a coverage element exists; I guess `IS NOT NULL` is good
> for that and sufficiently obvious not to need comment.

Yes, that was my reasoning, too.

> +This specification makes no guarantees on whether or not
> +\rtent{coverage} is retrievable (i.e., can be used in ADQL select
> The other new sections include the text  _"The ivoid column should
> be an explicit foreign key into resource."_ - should that be here
> too?

Oh, yes, it should.

Thanks for the review!

Changes are in commit 5d47fd9

mbtaylor commented:
I don't know if this is getting too clever, but I just thought of something else: the xpaths for `time_start`/`spectral_start` and `time_end`/`spectral_end` could be `substring-before(., ' ')` and `substring-after(., ' ')`.  I haven't attempted to test those, but I think they ought to do the right thing.  What do you think?

msdemlei commented:
On Thu, Feb 03, 2022 at 06:05:08AM -0800, Mark Taylor wrote:
> I don't know if this is getting too clever, but I just thought of
> something else: the xpaths for `time_start`/`spectral_start` and
> `time_end`/`spectral_end` could be `substring-before(., ' ')` and
> `substring-after(., ' ')`.  I haven't attempted to test those, but
> I think they ought to do the right thing.  What do you think?

My first gut feeling is that it's getting too clever.  As is usual
with utypes, they're not much good for anything except perhaps a bit
of documentation, and for that entering xpath functions probably
doesn't buy anything I'd be willing to pay for.

Can you think of a situation where that would help anyone?

mbtaylor commented:
Yeah, you're probably right.

mbtaylor reviewed (approved):
Reposting approval following rebase.
